connector/sumconnector/config.go

```diff
@@ -24,3 +34,88 @@ type AttributeConfig struct {
 	Key          string `mapstructure:"key"`
 	DefaultValue any    `mapstructure:"default_value"`
 }
+
+func (c *Config) Validate() error {
```

Comment:
Do you think it would make sense to validate each context in its own method, then call each method from `Validate`? Would that be easier to follow, or more complicated?

Reply:
In this specific case I sort of like the "have it all there sequentially" format for the tests as it involves a bunch of tests so it minimizes jumping around in the IDE between methods. It also sort of seems to be common in some of the other connectors.

That said, I totally understand the ask and if you'd prefer it be split up I'm happy to oblige.